net/retriever/retriever.go

```diff
@@ -92,10 +93,16 @@ func (m *MessageRetriever) fetchPointers() {
 	// Wait for each goroutine to finish then process any remaining messages that needed
 	// to be processed last
 	wg.Wait()
-	for _, env := range m.messageQueue {
-		m.handleMessage(env, nil)
+
+DRAIN_LOOP:
+	for {
+		select {
+		case env := <-m.messageQueue:
+			m.handleMessage(env, nil)
+		default:
+			break DRAIN_LOOP
```

Comment:
Ok I see. This will probably work except it only gets to this point after the wait group finishes. But with the current code the goroutines don't call wg.Done() until the function exits. But the function wont exit due to `m.messageQueue <- env`. So I think this would deadlock.